shelley/chain-and-ledger/executable-spec/test/Test/Shelley/Spec/Ledger/Address.hs

```diff
@@ -81,19 +91,80 @@ goldenTests =
         "addrEnterpriseS"
         putAddr
         (Addr scriptHash StakeRefNull)
-        "7205060708"
+        "7205060708",
+
+      -- serialisation golden tests
+      goldenSerialisation
+        "addrEnterpriseK for network id = 0"
+        (Addr @RealisticCrypto (keyBlake2b224 paymentKey) StakeRefNull)
+        "608a4d111f71a79169c50bcbc27e1e20b6e13e87ff8f33edc3cab419d4",
+      goldenSerialisation
+        "addrBaseKK for network id = 0"
+        (Addr @RealisticCrypto (keyBlake2b224 paymentKey) (StakeRefBase (keyBlake2b224 stakeKey)))
+        "008a4d111f71a79169c50bcbc27e1e20b6e13e87ff8f33edc3cab419d408b2d658668c2e341ee5bda4477b63c5aca7ec7ae4e3d196163556a4",
+      goldenSerialisation
+        "addrPtrK for network id = 0"
+        (Addr @RealisticCrypto (keyBlake2b224 paymentKey) (StakeRefPtr ptr))
+        "408a4d111f71a79169c50bcbc27e1e20b6e13e87ff8f33edc3cab419d481000203"
+
+       -- the same as above for other network ids is also needed, for example, for 3, 6
     ]
 
 golden :: String -> (a -> B.Put) -> a -> LBS.ByteString -> TestTree
 golden name put value expected =
   T.testCase name $
     T.assertEqual name expected (LB16.encode . B.runPut . put $ value)
 
+-- helper data to mimick realistic crypto
+data RealisticCrypto
+
+instance Crypto RealisticCrypto where
+  type DSIGN RealisticCrypto = Ed25519DSIGN
+  type KES   RealisticCrypto = SimpleKES Ed25519DSIGN 14
+  type VRF   RealisticCrypto = SimpleVRF
+  type HASH  RealisticCrypto = Blake2b_256
+  networkMagicId _ = Mainnet
```

Comment:
of course I can introduce a number of RealisticCrypto with different values of Network here. But maybe I can do it within the same RealisticCrypto :thinking: